lib/modules/platform/gitea/index.ts

```diff
@@ -173,6 +180,47 @@ async function lookupLabelByName(name: string): Promise<number | null> {
   return labelList.find((l) => l.name === name)?.id;
 }
 
+function getRepoUrl(
+  repo: helper.Repo,
+  gitUrl: GitUrlOption | undefined
+): string {
+  if (gitUrl === 'ssh') {
+    if (!repo.ssh_url) {
+      throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+    }
+    logger.debug({ url: repo.ssh_url }, `using SSH URL`);
+    return repo.ssh_url;
+  }
+
+  // Find options for current host and determine Git endpoint
+  const opts: HostRule = hostRules.find({
+    hostType: PlatformId.Gitea,
+    url: defaults.endpoint,
+  });
+
+  if (gitUrl === 'endpoint' || repo.clone_url === null) {
+    if (repo.clone_url === null) {
+      logger.debug('No clone_url found. Falling back to old behaviour.');
+    }
+
+    const { protocol, host, pathname } = parseUrl(defaults.endpoint);
+    const newPathname = pathname.slice(0, pathname.indexOf('/api'));
```

Comment:
what happens when no `/api` in path?

Reply:
If the `/api/v1` path is missing in the endpoint URL, renovate does not work at all. The path is not appended by the code. Maybe we could add error handling to validate the given endpoint at some point? Perhaps we could add some ensureTrailingApiPath() logic to [initPlattform()](https://github.com/renovatebot/renovate/blob/01d2ef5d3fdfd593028ec6d4098f0f606474d224/lib/modules/platform/gitea/index.ts#L186)? For example checking for trailing `/api/v<number>` via regex: `/.+\/api\/v\d+/`.

Reply:
I think we should remove it from platform url and add on required api calls instead. That should be done in separate pr.

Reply:
yes, but this should be done in a separate PR.